#6549 - Apache Tika can not parse Microsoft Docx format in native mode

tpenakov opened the pull request:
Fixes: #6549
A few things to points out:

- xalan dependency is included - probably on wrong place: extensions/arc/runtime/pom.xml
- additional config property is added for supported file types - we have support for pptx ans xlsx file types as well, however if we include all of them - the native build ends with OOM error.
- Trying to use `@ConfigProperty` in `io.quarkus.it.tika.TikaEmbeddedContentTest` leads to NPE for native build. This one (#2061) claims that is fixed, but I am receiving it.

tpenakov committed:
#6549 - Apache Tika can not parse Microsoft Docx format in native mode

geoand commented on bom/runtime/pom.xml:
```diff
@@ -2762,6 +2775,32 @@
             </dependency>
+
+            <!-- reflections -->
+            <dependency>
```
I am pretty sure we don't want to add this dependency to the core. We have basically done without it for all this time, so I don't really think it's warranted to add it now.

tpenakov replied:
Hi @geoand - if you check the code the reflection framework is used in more than one place. I've just put it on more centralized place. Anyway - what I've tried to achieve here is to register for reflection all classes in given package. Could you please advice is there an other way to achieve it?

gastaldi replied:
We use Jandex instead to query the application classes structures, see the examples in other extensions

geoand replied:
@tpenakov like @gastaldi said, we usually don't use reflection in Quarkus extensions but instead rely on Jandex (which is effectively an index for Java code) to look up various metadata.

One question, do you really need to register all classes of a package for reflection?

tpenakov replied:
Hi @geoand ,

The package which I want to register contains the implementation of components for parsing docx format (org.openxmlformats.schemas.wordprocessingml.x2006.main.impl). Also will need to include all classes in packagess for parsing xlsx and pptx components. 
I've tried to google this for Jandex, but without success. If you know how to do it in Jandex (to register all classes for given package) - this will save me a lot of time and I will remove the reflection framework  :)

geoand replied:
I don't think Jandex does this TBH. If you must register all classes in a package for reflection, then sure go ahead and use the reflections module (if @sberyozkin agrees), but please only use it in your extension and no where else

tpenakov replied:
Hi @geoand , @gastaldi  and @sberyozkin,
Reflections maven artifact is now moved under the tika-deployment.

gastaldi commented on extensions/arc/runtime/pom.xml:
```diff
@@ -27,6 +27,17 @@
       <artifactId>microprofile-context-propagation-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>xalan</groupId>
```
Why ARC needs to depend on Xalan?

tpenakov replied:
Hi @gastaldi ,
If I put it on the other places the tika-integration-tests are failed.

gastaldi commented on core/deployment/src/main/java/io/quarkus/deployment/util/ReflectUtil.java:
```diff
@@ -1,22 +1,17 @@
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.lang.reflect.*;
```
We don't use star imports, make sure to use the IDE config file as described in https://github.com/quarkusio/quarkus/blob/master/CONTRIBUTING.md#ide-config-and-code-style

tpenakov replied:
You were right @gastaldi -the IDE config file wasn't included. Now it is, however the wildcard import is still there :(

gastaldi commented on core/deployment/pom.xml:
```diff
@@ -79,6 +79,10 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
As @geoand mentioned, I don't think it's a good idea to add this library to core

tpenakov replied:
Hi @gastaldi ,
 if you check the code the reflection framework is used in more than one place. I've just put it on more centralized place. Anyway - what I've tried to achieve here is to register for reflection all classes in given package. Could you please advice is there an other way to achieve it?

Not a problem to move it only on Tika related deploynment, but as I said this library is used on more than one place. In my experience if such things is happened, we have to move the dependency on more centralized place, or to not used it at all and to replace it with the similar functionality.

gastaldi commented on extensions/tika/runtime/pom.xml:
```diff
@@ -39,6 +39,20 @@
+                    <groupId>org.apache.xmlbeans</groupId>
+                    <artifactId>xmlbeans</artifactId>
+                </exclusion>
+            </exclusions>
```
You can remove this exclusion and the dependency below (since it's resolved transitively)

tpenakov replied:
Hi @gastaldi - There was a difference in the versions of XML Beans. I found only this way to deal with it.

sberyozkin commented:
@tpenakov Thanks for giving it a go, appreciated. I agree though with @geoand and @gastaldi is that the way it is done it would cause to many side-effects. Let me try to find some time asap and experiment with your PR locally, I'll keep you uptodate, thank you !

tpenakov commented:
Thank you @sberyozkin - I've added some comments/answers to @geoand  and @gastaldi and added and the explanations in the PR. I also have some concerns (written in PR description)

tpenakov committed:
#6549 - Apache Tika can not parse Microsoft Docx format in native mode - move reflections maven artifact under the tika-deployment module

gsmet commented:
Closing this one as out of date. Let's create a new PR if we still need it.
